RecoVertex/Configuration/python/RecoVertex_cff.py

```diff
@@ -33,36 +33,47 @@
                           )
 vertexreco = cms.Sequence(vertexrecoTask)
 
-#timing
-from RecoVertex.PrimaryVertexProducer.TkClusParameters_cff import DA2D_vectParameters
-DA2D_vectParameters.TkDAClusParameters.verbose = cms.untracked.bool(False)
-unsortedOfflinePrimaryVertices4D = unsortedOfflinePrimaryVertices.clone( verbose = cms.untracked.bool(False),
-                                                                         TkClusParameters = DA2D_vectParameters )
-unsortedOfflinePrimaryVertices4D.TkFilterParameters.minPt = cms.double(0.0)
-unsortedOfflinePrimaryVertices4D.TrackTimesLabel = cms.InputTag("trackTimeValueMapProducer:generalTracksConfigurableFlatResolutionModel")
-unsortedOfflinePrimaryVertices4D.TrackTimeResosLabel = cms.InputTag("trackTimeValueMapProducer:generalTracksConfigurableFlatResolutionModelResolution")
-offlinePrimaryVertices4D=sortedPrimaryVertices.clone(vertices="unsortedOfflinePrimaryVertices4D", particles="trackRefsForJetsBeforeSorting4D", trackTimeTag=cms.InputTag("trackTimeValueMapProducer","generalTracksConfigurableFlatResolutionModel"),trackTimeResoTag=cms.InputTag("trackTimeValueMapProducer","generalTracksConfigurableFlatResolutionModelResolution"),assignment=dict(useTiming=True))
-offlinePrimaryVertices4DWithBS=sortedPrimaryVertices.clone(vertices="unsortedOfflinePrimaryVertices4D:WithBS", particles="trackRefsForJetsBeforeSorting4D", trackTimeTag=cms.InputTag("trackTimeValueMapProducer","generalTracksConfigurableFlatResolutionModel"),trackTimeResoTag=cms.InputTag("trackTimeValueMapProducer","generalTracksConfigurableFlatResolutionModelResolution"),assignment=dict(useTiming=True))
-
-trackWithVertexRefSelectorBeforeSorting4D = trackWithVertexRefSelector.clone(vertexTag="unsortedOfflinePrimaryVertices4D")
-trackWithVertexRefSelectorBeforeSorting4D.ptMax=9e99
-trackWithVertexRefSelectorBeforeSorting4D.ptErrorCut=9e99
-trackRefsForJetsBeforeSorting4D = trackRefsForJets.clone(src="trackWithVertexRefSelectorBeforeSorting4D")
-
-from SimTracker.TrackerHitAssociation.tpClusterProducer_cfi import tpClusterProducer
-from SimTracker.TrackAssociatorProducers.quickTrackAssociatorByHits_cfi import quickTrackAssociatorByHits
-from SimTracker.TrackAssociation.trackTimeValueMapProducer_cfi import trackTimeValueMapProducer
+#modifications for timing
+from RecoVertex.Configuration.RecoVertex_phase2_timing_cff import *
 _phase2_tktiming_vertexrecoTask = cms.Task( vertexrecoTask.copy() ,
                                             tpClusterProducer ,
```

Comment:
I think that this ```_phase2_tktiming_vertexrecoTask``` (less the default ```vertexrecoTask.copy()```) together with ```_phase2_tktiming_layer_vertexrecoTask``` can be moved to the phase2 cff file; perhaps name them as ```tkTimingSpecific``` to indicate that they are just additions.

Then below all you need is to import these additions and do  ```phase2_timing.toReplaceWith(vertexrecoTask, cms.Task(vertexrecoTask.copy(),_phase2_tktimingSpecific_vertexrecoTask))```

Reply:
The other modifications would still require importing unsortedOfflinePrimaryVertices4Dfastsim, etc though

Reply:
And trying to split it this way I still get errors like:
"  File "/cvmfs/cms.cern.ch/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_4_0_mtd3/python/FWCore/ParameterSet/SequenceVisitors.py", line 116, in enter
    self._names.append(self._moduleToName[node])
KeyError: cms.EDProducer("PrimaryVertexProducer",
"

Reply:
that's fine,
in a similar way as the ```_phase2_tktimingSpecific_vertexrecoTask``` will have to be imported from the other file. One just wouldn't see how they are made